service/history/replication/batchable_task.go

```diff
@@ -158,6 +163,9 @@ func (w *batchedTask) handleIndividualTasks() {
 }
 
 func (w *batchedTask) AddTask(incomingTask TrackableExecutableTask) bool {
+	if w.state != batchStateOpen {
+		return false
+	}
```

Comment:
lock?

Reply:
This is a minor optimization. batchStateOpen is a one way switch. It can only change from `true` to `false`. So if batchStateOpen is `false`,  we can safely return false without locking.

Reply:
this kind of "optimization" makes no sense

1. prove that w.state can be safely accessed and the value will not break the logic here
2. this task is not going to be accessed by many goroutine, so how much nanoseconds will be saved? does this worth the complexity & potential data corruption?
3. use sync.Mutex or atomics or go chan from golang, do not rely on implicit assumption of any platform unless you write a ton of tests validating your impl is correct